openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -36,3 +46,52 @@ class ExternalId(TimeStampedModel):
     external_id_type = models.ForeignKey(ExternalIdType, db_index=True, on_delete=models.CASCADE)
     user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
     history = HistoricalRecords()
+
+    class Meta(object):
+        app_label = "external_id"
+
+    @classmethod
+    def user_has_external_id(cls, user, type_name):
+        """
+        Checks if a user has an ExternalId of the type_name provided
+        Arguments:
+            user: User to search for
+            type_name (str): Name of the type of ExternalId
+        """
+        try:
+            type_obj = ExternalIdType.objects.get(name=type_name)
+        except ExternalIdType.DoesNotExist:
+            LOGGER.info('No external id type of {}'.format(type_name))
+            return False
+        try:
+            external_id = cls.objects.get(
+                user=user,
+                external_id_type=type_obj
+            )
+            if external_id:
```

Comment:
Is this check needed? I.e. is there a case where `cls.objects.get` returns a False-y value instead of raising the exception?